symfony/plugins/orangehrmPimPlugin/Api/EmpUsTaxExemptionAPI.php

```diff
@@ -0,0 +1,241 @@
+        return new EndpointResourceResult(
+            EmpUsTaxExemptionModel::class, $saveUsTaxExemption,
+            new ParameterBag(
+                [CommonParams::PARAMETER_EMP_NUMBER => $saveUsTaxExemption->getEmployee()->getEmpNumber()]
+            )
+        );
+    }
+
+    /**
+     * @return EmpUsTaxExemption
+     */
+    private function saveTaxExemption(): EmpUsTaxExemption
+    {
+        $empNumber = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_EMP_NUMBER
+        );
+        $federalStatus = $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_FEDERAL_STATUS);
+        $federalExemptions = $this->getRequestParams()->getIntOrNull(RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_FEDERAL_EXEMPTIONS);
+        $taxStateCode = $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_TAX_STATE_CODE);
+        $stateStatus = $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_STATE_STATUS);
+        $stateExemptions = $this->getRequestParams()->getIntOrNull(RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_STATE_EXEMPTIONS);
+        $unemploymentStateCode = $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_UNEMPLOYMENT_STATE_CODE);
+        $workStateCode = $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_WORK_STATE_CODE);
+        $empUsTaxExemption = $this->getEmpUsTaxExemptionService()->getEmpUsTaxExemptionDao()->getEmployeeTaxExemption($empNumber);
+        if (!$empUsTaxExemption instanceof EmpUsTaxExemption) {
+            $empUsTaxExemption = new EmpUsTaxExemption();
+            $empUsTaxExemption->getDecorator()->setEmployeeByEmpNumber($empNumber);
+        }
+        $empUsTaxExemption->setFederalStatus($federalStatus);
+        $empUsTaxExemption->setFederalExemptions($federalExemptions);
+        $empUsTaxExemption->setState($taxStateCode);
+        $empUsTaxExemption->setStateStatus($stateStatus);
+        $empUsTaxExemption->setStateExemptions($stateExemptions);
+        $empUsTaxExemption->setUnemploymentState($unemploymentStateCode);
+        $empUsTaxExemption->setWorkState($workStateCode);
+        return $this->getEmpUsTaxExemptionService()->getEmpUsTaxExemptionDao()->saveEmployeeTaxExemption($empUsTaxExemption);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForUpdate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_EMP_NUMBER,
+                new Rule(Rules::IN_ACCESSIBLE_EMP_NUMBERS)
+            ),
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+                new Rule(Rules::BETWEEN, [0, 100])
```

Comment:
rule is wrong. Since id hardcoded in route

Reply:
added equal [0]